File: src/main/java/com/antlr/plugin/psi/ChannelSpecNode.java
package com.antlr.plugin.psi;

import com.intellij.lang.ASTNode;
import com.intellij.psi.tree.IElementType;
import com.intellij.psi.util.PsiTreeUtil;
import com.antlr.plugin.ANTLRv4TokenTypes;
import com.antlr.plugin.parser.ANTLRv4Lexer;
import org.jetbrains.annotations.NotNull;

/**
 * A channel defined in the {@code channels} section.
 */
public class ChannelSpecNode extends RuleSpecNode {

    public ChannelSpecNode(@NotNull ASTNode node) {
        super(node);
    }

    @Override
    public GrammarElementRefNode getNameIdentifier() {
        return PsiTreeUtil.getChildOfType(this, LexerRuleRefNode.class);
    }

    @Override
    public IElementType getRuleRefType() {
        return ANTLRv4TokenTypes.TOKEN_ELEMENT_TYPES.get(ANTLRv4Lexer.TOKEN_REF);
    }
}

File: src/main/java/com/antlr/language/psi/ChannelSpecNode.kt
package com.antlr.language.psi

import com.antlr.language.ANTLRv4Lexer
import com.antlr.language.AntlrTokenTypes
import com.intellij.lang.ASTNode
import com.intellij.psi.tree.IElementType
import com.intellij.psi.util.PsiTreeUtil

/**
 * A channel defined in the `channels` section.
 */
class ChannelSpecNode(node: ASTNode) : RuleSpecNode(node) {
    public override fun getNameIdentifier(): GrammarElementRefNode? {
        return PsiTreeUtil.getChildOfType(this, LexerRuleRefNode::class.java)
    }

    override val ruleRefType: IElementType?
        get() = AntlrTokenTypes.TOKEN_ELEMENT_TYPES?.get(ANTLRv4Lexer.TOKEN_REF)
}
